tests/test_generic.py

```diff
@@ -1247,6 +1247,7 @@ def test_encodedstream_set_data():
     assert cc[NameObject("/Test")] == "/MyTest"
 
 
+@pytest.mark.enable_socket()
```

Comment:
Why was this test not failing before?

I thought that forgetting to enable_socket would cause it to fail when the test tries to download something. Did that change? Is maybe caching the issue? But then the first execution should never have worked, right?

Reply:
This has been a side effect. Running the tests for version 4.1.0 showed the following behavior:

* Running `pytest -k 'test_calling_indirect_objects'` locally fails correctly.
* Running `pytest` locally does not fail.

It turns out that due to https://github.com/py-pdf/pypdf/blob/f43268734a529d4098e6258bf346148fd24c54f0/tests/test_generic.py#L1203-L1216, the file would already be available with the regular execution order and thus we would not see this fail in CI.